xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/test/java/org/xwiki/rest/internal/ModelFactoryTest.java

```diff
@@ -0,0 +1,142 @@
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.net.URI;
+import java.util.List;
+
+import javax.inject.Provider;
+
+@ComponentTest
+public class ModelFactoryTest
+{
+    protected XWikiContext xcontext = mock(XWikiContext.class);
+    
+    @InjectComponentManager
+    protected ComponentManager componentManager;
+
+    @MockComponent
+    protected Provider<XWikiContext> xcontextProvider;
+
+    @MockComponent
+    SpaceReferenceResolver<String> resolver;
+    
+    @InjectMockComponents
+    ModelFactory modelFactory;
+    
+    @Test
+    @SuppressWarnings("deprecation")
+    void testToRestObject() throws Exception
+    {
+        when(xcontextProvider.get()).thenReturn(xcontext);
+        
+        BaseObject xwikiObject = mock(BaseObject.class);
+        BaseClass xwikiClass  = mock(BaseClass.class);
+       
+        when(xwikiObject.getPropertyNames()).thenReturn(new String[] {});
+        when(xwikiObject.getXClass(xcontext)).thenReturn(xwikiClass);
+        when(xwikiObject.getClassName()).thenReturn("Some.XClass");
+        when(xwikiObject.getNumber()).thenReturn(0);
+        
+        StringClass stringField = new StringClass();
+        stringField.setName("textValue");
+        StringProperty textElement = new StringProperty();
+        textElement.setName("textValue");
+        textElement.setClassType("String");
+        textElement.setValue("abcd");
+        when(xwikiObject.get("textValue")).thenReturn(textElement);
+
+        PasswordClass pwField = new PasswordClass();
+        pwField.setName("passwordValue");
+        StringProperty pwElement = new StringProperty();
+        pwElement.setName("passwordValue");
+        pwElement.setClassType("Password");
+        pwElement.setValue("abcd");
+        when(xwikiObject.get("passwordValue")).thenReturn(pwElement);
+        
+        when(xwikiClass.getProperties()).thenReturn(new java.lang.Object[] {stringField, pwField});
+        
+        // some more mockery needed to pass the "toRestObject" call
+        // although the results are not tested (yet)
+
+        URI dummy = new URI("https://localhost/");
+        Document doc = mock(Document.class);
+        SpaceReference spaceRef = new SpaceReference("Space", new WikiReference("wiki"));
+        when(doc.getPrefixedFullName()).thenReturn("wiki:Space.Page");
+        when(doc.getWiki()).thenReturn("wiki");
+        when(doc.getSpace()).thenReturn("Space");
+        when(doc.getName()).thenReturn("Page");
+        when(doc.getDocumentReference()).thenReturn(new DocumentReference("Page", spaceRef));
+        when(resolver.resolve(eq("Space"), any(String.class))).thenReturn(spaceRef);
+        
+        com.xpn.xwiki.web.Utils.setComponentManager(componentManager);
```

Comment:
Actually all this seems to indicate you should use `@OldcoreTest` instead of `@ComponentTest`.  That could allow you to reduce the setup code.

Reply:
ok, trying this ....